File: week4/q7.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None
def is_circular(head):
    if not head:
        return True
    temp = head.next
    while temp and temp != head:
        temp = temp.next
    return temp == head
head = Node(1)
head.next = head
print(is_circular(head))

File: week4_C/q7.c
#include <stdio.h>
#include <stdlib.h>
struct Node { int data; struct Node* next; };
int isCircular(struct Node* head) {
    if (!head) return 1;
    struct Node* temp = head->next;
    while (temp && temp != head)
        temp = temp->next;
    return temp == head;
}
int main() {
    struct Node* head = malloc(sizeof(struct Node));
    head->next = head;
    printf("%d", isCircular(head));
}